nixos/modules/services/web-apps/wallabag.nix

```diff
@@ -0,0 +1,190 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.wallabag;
+
+  poolName = "wallabag";
+
+  phpfpmSocketName = "/var/run/phpfpm/${poolName}.sock";
+
+  configFile = pkgs.writeTextFile {
+    name = "wallabag-config";
+    text = cfg.conf;
+    destination = "/config/parameters.yml";
+  };
+
+  appDir = pkgs.buildEnv {
+    name = "wallabag-app-dir";
+    ignoreCollisions = true;
+    checkCollisionContents = false;
+    paths = [ configFile "${cfg.package}/app" ];
+  };
+
+in {
+  options = {
+    services.wallabag = {
+      enable = mkEnableOption "wallabag";
+
+      user = mkOption {
+        type = types.str;
+        default = "nginx";
+        description = ''
+          User account under which both the update daemon and the web-application run.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/wallabag";
+        description = ''
+          Data directory.
+        '';
+      };
+
+      package = mkOption {
```

Comment:
I think the low-grade consensus we currently have is that if there are not multiple versions of the package already in nixpkgs that could be placed here, then it's better not to expose a `package` option, for eval-time reasons.

AFAIK there is a single `wallabag` package in nixpkgs, so maybe this option should go away.

Reply:
Forget about this, it looks like it's not that bad after all https://github.com/NixOS/nixpkgs/issues/57477